onnxruntime/core/framework/utils.cc

```diff
@@ -568,7 +568,7 @@ int32_t ONNXTensorElementDataTypeToProtoTensorType(ONNXTensorElementDataType onn
   }
 }
 
-#ifdef ENABLE_TRAINING
+#if defined(ENABLE_TRAINING) || defined(ENABLE_TRAINING_OPS)
```

Comment:
Ops that need Contiguous inputs are NcclAllReduce, NcclReduceScatter, NcclAllGather... 
I don't think it need defined(ENABLE_TRAINING_OPS) here.